platform/security/platform-security-core-api/src/main/java/ddf/security/samlp/SamlProtocol.java

```diff
@@ -489,9 +492,18 @@ public static EntityDescriptor createSpMetadata(
 
     entityDescriptor.getRoleDescriptors().add(spSsoDescriptor);
 
+    entityDescriptor.setCacheDuration(getCacheDuration());
+
     return entityDescriptor;
   }
 
+  static long getCacheDuration() {
+    return Long.parseLong(
```

Comment:
✏️ Referring back to my human-readable suggestion for the system property, that would make this look something like...
```java
return Duration.ofDays(Long.parseLong(System.getProperty(
        "org.codice.ddf.security.saml.Metadata.cacheDuration",
        "7"))).toMillis();
```